action/protocol/staking/handler_stake_migrate.go

```diff
@@ -181,11 +181,14 @@ func (p *Protocol) ConstructExecution(ctx context.Context, act *action.MigrateSt
 	}
 	duration := uint64(bucket.StakedDuration / p.helperCtx.BlockInterval(protocol.MustGetBlockCtx(ctx).BlockHeight))
 
-	return p.constructExecution(candidate.GetIdentifier(), bucket.StakedAmount, duration, nonce, gas, gasPrice)
+	return p.constructExecution(ctx, candidate.GetIdentifier(), bucket.StakedAmount, duration, nonce, gas, gasPrice)
 }
 
-func (p *Protocol) constructExecution(candidate address.Address, amount *big.Int, duration uint64, nonce uint64, gasLimit uint64, gasPrice *big.Int) (action.Envelope, error) {
-	contractAddress := p.config.MigrateContractAddress
+func (p *Protocol) constructExecution(ctx context.Context, candidate address.Address, amount *big.Int, duration uint64, nonce uint64, gasLimit uint64, gasPrice *big.Int) (action.Envelope, error) {
+	contractAddress := p.config.TimestampedMigrateContractAddress
+	if !protocol.MustGetFeatureCtx(ctx).TimestampedStakingContract {
+		contractAddress = p.config.MigrateContractAddress
+	}
```

Comment:
after this hard-fork, we do not allow current v2 contract bucket to migrate?

Reply:
the `StakeMigrate` action is to migrate native bucket to contract v2 bucket originally, and now will migrate to contract v3 bucket.

Reply:
should be: verify func/call flow works for v3 contract

Reply:
good reminder